src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -1264,9 +1225,6 @@ public override bool Equals([NotNullWhen(true)] object? obj)
                     return Unsafe.As<byte, ulong>(ref pThisValue) == Unsafe.As<byte, ulong>(ref pOtherValue);
 
 #if RARE_ENUMS
-                case CorElementType.ELEMENT_TYPE_BOOLEAN:
```

Comment:
Similar for GetHashCode, Equals, Compare.

The optimizations make assumptions (e.g. here: https://github.com/dotnet/runtime/blob/a78c46d1dd9ad718505e3b0508e2c72d6fca6b21/src/coreclr/vm/jitinterface.cpp#L4879-L4882) that these method always call the underlying type.